src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -95923,5 +95947,306 @@ void M()
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, @"new[] { new[] { ""string"" , null } }").WithArguments("string?[][]", "System.Collections.Generic.IEnumerable<System.Collections.Generic.IEnumerable<string>>").WithLocation(11, 47)
                 );
         }
+
+        [Fact]
+        [WorkItem(30563, "https://github.com/dotnet/roslyn/issues/30563")]
+        public void ExtensionMethodDelegate_01()
+        {
+            var source =
+@"delegate void D0();
+delegate void D1<T>(T t);
+static class E
+{
+    internal static void F<T>(this T t) { }
+}
+class Program
+{
+    static void M0(string? x, string y)
+    {
+        D0 d;
+        d = x.F;
+        d = x.F<string?>;
+        d = x.F<string>; // 1
+        d = y.F;
+        d = y.F<string?>;
+        d = y.F<string>;
+    }
+    static void M1()
+    {
+        D1<string?> d;
+        D1<string> e;
+        d = E.F<string?>;
+        d = E.F<string>; // 2
+        e = E.F<string?>;
+        e = E.F<string>;
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            // https://github.com/dotnet/roslyn/issues/30563: Should not report "CS8602: Possible dereference" for x.F.
+            comp.VerifyDiagnostics(
+                // (12,13): warning CS8602: Dereference of a possibly null reference.
```

Comment:
Is this text correct? I see a `// 1` on a different line.

Reply:
The incorrect warning on several of these tests with method group conversions (rather than delegate creation expressions) is due to [https://github.com/dotnet/roslyn/issues/30563](https://github.com/dotnet/roslyn/issues/30563). I'm working on that issue currently but the fix may be more involved so I wanted to address it separately.

---
In reply to: [281867959](https://github.com/dotnet/roslyn/pull/35572#discussion_r281867959) [](ancestors = 281867959)